code/game/objects/items/crayons.dm

```diff
@@ -729,20 +729,13 @@
 
 		return
 
-	if(isobj(target) && !istype(target, /obj/effect/decal/cleanable/crayon/gang))
+	if(istype(target, /obj/structure/window))
```

Comment:
Do something like this:
```cpp
if(actually_paints && target.flags_1 & CAN_BE_PAINTED_1)
	stuff
```
You can substitute the flag for a redefined proc or whatever.
In short, add a simple way to make things able to be painted or not, instead of having potentially a million istype checks.
That will give you finer control on what can be painted and what cannot be.

Reply:
we already have a component to make things paintable I'm just reverting what made it obsolete

https://github.com/tgstation/tgstation/blob/add74bee56cfde91602eb8ee28e3cae5959bc4f6/code/datums/components/paintable.dm